File: app/src/main/java/com/yeputra/moviecatalogue/ui/TVShowFm.java
package com.yeputra.moviecatalogue.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.yeputra.moviecatalogue.R;
import com.yeputra.moviecatalogue.adapter.MovieAdapter;
import com.yeputra.moviecatalogue.base.BaseFragment;
import com.yeputra.moviecatalogue.model.Movie;
import com.yeputra.moviecatalogue.presenter.DataPresenter;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
import butterknife.BindView;
import butterknife.ButterKnife;

public class TVShowFm extends BaseFragment<DataPresenter> {
    @BindView(R.id.swiperefresh)
    SwipeRefreshLayout swipeRefreshLayout;

    @BindView(R.id.list_item)
    RecyclerView rvItem;

    private MovieAdapter adapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_movie, container, false);
        ButterKnife.bind(this, v);
        return v;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        swipeRefreshLayout.setColorSchemeColors(ContextCompat.getColor(getContextView(),R.color.colorAccent));
        swipeRefreshLayout.setOnRefreshListener(() -> presenter.getMovies());

        adapter = new MovieAdapter(movie -> {

        });
        rvItem.setLayoutManager(new LinearLayoutManager(getContextView()));
        rvItem.setOverScrollMode(View.OVER_SCROLL_NEVER);
        rvItem.setAdapter(adapter);
        presenter.getTVMovies();
    }

    @Override
    public void onPresenterSuccess(Object data) {
        super.onPresenterSuccess(data);
        adapter.replaceItem((List<Movie>) data);
    }

    @Override
    protected DataPresenter initPresenter() {
        return new DataPresenter(this);
    }

    @Override
    public void showProgressbar() {
        swipeRefreshLayout.setRefreshing(true);
    }

    @Override
    public void hideProgressbar() {
        swipeRefreshLayout.setRefreshing(false);
    }

}

File: app/src/main/java/com/yeputra/moviecatalogue/ui/TVShowFm.kt
package com.yeputra.moviecatalogue.ui

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.GridLayoutManager
import com.yeputra.moviecatalogue.R
import com.yeputra.moviecatalogue.adapter.MovieAdapter
import com.yeputra.moviecatalogue.base.BaseFragment
import com.yeputra.moviecatalogue.model.MovieResponse
import com.yeputra.moviecatalogue.utils.Constans
import com.yeputra.moviecatalogue.viewmodel.TVViewModel
import kotlinx.android.synthetic.main.fragment_movie.*

class TVShowFm : BaseFragment<TVViewModel>() {
    private lateinit var adapter: MovieAdapter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_movie, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        onShowProgressbar()

        swiperefresh.setColorSchemeColors(ContextCompat.getColor(contextView, R.color.colorAccent))
        swiperefresh.setOnRefreshListener {
            onShowProgressbar()
            viewmodel.getTVShow().observe(this, setTVShow)
        }

        adapter = MovieAdapter {
            val intent = Intent(contextView, DetailMovieActivity::class.java)
            intent.putExtra(Constans.INTENT_DATA, it)
            startActivity(intent)
        }

        list_item.layoutManager = GridLayoutManager(contextView, 2)
        list_item.overScrollMode = View.OVER_SCROLL_NEVER
        list_item.adapter = adapter
        viewmodel.getTVShow().observe(this, setTVShow)
    }

    private val setTVShow = Observer<MovieResponse> {
        it.results?.let { it1 -> adapter.setItem(it1) }
        onHideProgressbar()
    }

    override fun initPresenter(): TVViewModel {
        return ViewModelProviders.of(this).get(TVViewModel::class.java)
    }

    override fun onShowProgressbar() {
        swiperefresh.isRefreshing = true
    }

    override fun onHideProgressbar() {
        swiperefresh.isRefreshing = false
    }

}